mule-http-connector/src/test/java/org/mule/test/http/api/HttpRequestAttributesSerializationTestCase.java

```diff
@@ -68,16 +84,23 @@ public void withNoCertificate() {
 
   @Test
   @Description("HttpRequestAttributes are correctly serialized and deserialized with an explicit certificate. Certificate can be recover after deserialization")
-  public void withResolvedCertificate() {
-    HttpRequestAttributes processed = assertSerialization(baseBuilder.clientCertificate(certificate).build());
-    assertThat(processed.getClientCertificate(), is(certificate));
+  public void withResolvedCertificate() throws Exception {
+    HttpRequestAttributes processed =
+        assertSerialization(baseBuilder.clientCertificate(buildCertificateData(certificate)).build());
+    assertThat(processed.getClientCertificate().toString(), is(certificate.toString()));
```

Comment:
Why should we change this assertion? Is the equals() method working?

Reply:
Instead of using toString(), a test utility method areSameCertificate was added to compare Certificate and CertificateData.